app/src/main/java/com/example/chrono/main/circuit/CircuitTimerActivity.kt

```diff
@@ -0,0 +1,163 @@
+package com.example.chrono.main.circuit
+
+import android.app.Activity
+import android.media.AsyncPlayer
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.CountDownTimer
+import android.view.View
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
+import com.example.chrono.R
+import com.example.chrono.databinding.ActivityCircuitTimerBinding
+import com.example.chrono.util.objects.CircuitObject
+import com.google.gson.GsonBuilder
+import kotlin.math.roundToInt
+
+class CircuitTimerActivity : AppCompatActivity() {
+    private var bind: ActivityCircuitTimerBinding? = null
+
+    enum class TimerState { INIT, RUNNING, PAUSED }
+    enum class RunningState { INIT, WORK, REST }
+
+    private lateinit var countdown: CountDownTimer
+    private var timerState: TimerState = TimerState.INIT
+    private lateinit var runningState: RunningState
+
+    private var secondsLeft: Float = 0.0f
+    private var currentSet: Int = 0
+
+    private var sets: Int = 0
+    private var timeRest: Int = 0
+    private var timeWork: Int = 0
+
+//    private lateinit var audioPlayer: AsyncPlayer
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_circuit_timer)
+        bind = DataBindingUtil.setContentView(this, R.layout.activity_circuit_timer)
+
+        loadTimer(
+            GsonBuilder().create()
+                .fromJson(intent.getStringExtra("circuitObject"), CircuitObject::class.java)
+        )
+
+        // Initialize stuff perhaps
+
+        bind!!.startbutton.setOnClickListener {
+            workout()
+            timerState = TimerState.RUNNING
+            updateButtonUI()
+        }
+
+        bind!!.pausebutton.setOnClickListener {
+            countdown.cancel()
+            timerState = TimerState.PAUSED
+            updateButtonUI()
+        }
+
+        bind!!.resumebutton.setOnClickListener {
+            createTimer(secondsLeft.toInt(), true)
+            timerState = TimerState.RUNNING
+            updateButtonUI()
+        }
+
+        bind!!.closeButton.setOnClickListener {
+            // Make sure that the timer is shut down
+            countdown.cancel()
+            setResult(Activity.RESULT_CANCELED)
+            finish()
+        }
+    }
+
+    private fun createTimer(seconds: Int, wasPaused: Boolean) {
+        var time = seconds.toFloat() * 1000 + 250
+        if (wasPaused) {
+            time = secondsLeft * 1000 + 250
+        }
+        countdown = object : CountDownTimer(time.toLong(), 250) {
+            override fun onTick(p0: Long) {
```

Comment:
I think onTick occurs every 250 ms and the timer remaining textView also updates every 250ms even though we are only showing remaining time to 1 second. Is what I'm thinking correct?

Reply:
Yeah so onTick will be called every 250 ms but the countdown timer itself is going down only every second.